comp110-worksheet-6/OxoBoard.cs

```diff
@@ -8,69 +8,380 @@ namespace comp110_worksheet_6
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+        private Grid[,] grids;
+        private int inARow = 3;
+        private string horizontalASCII = "--+---+--";
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+
+        /// <summary>
+        /// The game constructor which assigns the grid size as well as the amount 
+        /// in a row for a player to win.
+        /// </summary>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        /// <param name="inARow"></param>
+        public OxoBoard(int width = 3, int height = 3, int inARow = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            grids = new Grid[width, height];
+            this.inARow = inARow;
+
+            PopulateGrid();
+            horizontalASCII = GetHorizontalString();
 		}
 
-		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
+        /// <summary>
+        /// Populates the multidimensional array storing the grids class
+        /// and instantiates a new class and assigns the appropriate variables
+        /// </summary>
+        private void PopulateGrid()
+        {
+            for (int x = 0; x < grids.GetLength(0); x++)
+            {
+                for (int y = 0; y < grids.GetLength(1); y++)
+                {
+                    grids[x, y] = new Grid(Mark.None, new Tuple<int, int>(x, y));
+                    AssignWinningPositions(grids[x, y]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Used to cycle through each Grid's possible combinations and then cycles through
+        /// those combinations to see if they all match one specific mark
+        /// </summary>
+        /// <param name="gridTile"></param>
+        /// <returns>
+        /// If a combination has been met then it returns the mark that won otherwise it
+        /// returns none which states no one has one yet
+        /// </returns>
+        private Mark CheckGridTile(Grid gridTile)
+        {
+            List<Mark> marks;
+
+            foreach (var winningPositions in gridTile.winPositions)
+            {
+                marks = new List<Mark>();
+                for (int i = 0; i < winningPositions.Count; i++)
+                { 
+                    if (winningPositions[i].Item1 < grids.GetLength(0) && winningPositions[i].Item2 < grids.GetLength(1))
+                    {
+                        marks.Add(GetSquare(winningPositions[i].Item1, winningPositions[i].Item2));
+                    }
+                    
+                }
+                
+                if (IsAllTheSameMarks(marks.ToArray()) && marks.Count == inARow)
```

Comment:
Another efficiency comment: change `IsAllTheSameMarks` to take a `List`, to avoid having to convert the list to an array